### ISSUE TITLE
No resource or method named `shell_out!' for `Chef::Provider::SelinuxPolicyFcontext ""

### ISSUE THREAD

**Issue by jfrome** [BackSlasher/chef-selinuxpolicy#7]
I am receiving the following error when attempting to use this cookbook:

```
NoMethodError
execute[selinux-fcontext-httpd_sys_script_exec_t-addormodify] (/var/chef/cache/cookbooks/selinux-policy/providers/fcontext.rb line 37) had an error: NoMethodError: No resource or method named `shell_out!' for `Chef::Provider::SelinuxPolicyFcontext ""'
```

This is what the compiled resource looks like in the debug log:
```
    # Declared in /var/chef/cache/cookbooks/ep-nagios/recipes/thruk.rb:90:in `from_file'
    
    selinux_policy_fcontext("/usr/share/thruk/fcgid_env.sh") do
      action :addormodify
      retries 0
      retry_delay 2
      guard_interpreter :default
      cookbook_name "ep-nagios"
      recipe_name "thruk"
      secontext "httpd_sys_script_exec_t"
      file_spec "/usr/share/thruk/fcgid_env.sh"
    end
```

Here is the snippet where it is called in the recipe:
```
selinux_policy_fcontext '/usr/share/thruk/fcgid_env.sh' do
  secontext 'httpd_sys_script_exec_t'
end
```

More details:

1. OS: Centos6.6
2. Chef client: 11.18.6
3. selinux is enforcing and targeted.
4. semanage setsebool and getsebool all work when called from the shell

Any ideas?

**Comment by BackSlasher** [comment]
Please try something like:
```ruby
require 'mixin/shell_out'
include Chef::Mixin::Shellout
```
in `libraries/helper-disabled.rb` and see if this helps.
This is the only place I use shellout and it works in Chef 12 - might be a Chef 11 issue.
If this helps I'll put it in master.

**Comment by Sauraus** [comment]
Upgrade to Chef client 12.x and the problem goes away. :)

**Comment by BackSlasher** [comment]
I want to support Chef 11 (AWS Opsworks uses that version IMO) if possible and easy.

@jfrome please check and LMK when convinient

**Comment by jfrome** [comment]
PR #9 fixes the issue for me.

**Comment by BackSlasher** [comment]
Should be solved. Please reopen/lmk in some other way if not

**Comment by thiagonache** [comment]
Hi there,

I'm using aws opsworks with chef 11.10 and I'm facing this problem.

**Comment by BackSlasher** [comment]
Should be solved in version `0.6.2` of cookbook and above. If you have that version (or higher) and still experience it, please attach the error message

**Comment by thiagonache** [comment]
it was out of date.
thank you
